### DISCUSSION TITLE
Did anyone play around with AstroJS and this library?

### DISCUSSION THREAD

**Discussion by dohomi** [garronej/tss-react#184] [General]
Hey Guys

I'd like to know if anyone successfully integrated tss-react with AstroJS? As there seem to be already a path for the new app directory for NextJS it would be interesting to know if a similar approach can be taken to make it work with AstroJS as well

Cheers

**Comment by garronej** [comment]
Hello @dohomi,  

Despite the fact that I do not yer provide any utility for Astro it's good to know that `tss-react` is build on top of [emotion](https://emotion.sh).  MUI is using emotion internally.  

So if you make either Emotion or MUI work, you make TSS work as well.  

I found this answer that very much look like might work: https://github.com/withastro/astro/issues/4432  

If you have any success with it, coud you report to me, I'll include an helper.  

Many thanks

**Reply by dohomi** [reply]
I found that issue as well, but its only a partly solution - which is totally not fine for a SSG website (especially if FOUC would be implemented). There is very little info out to make it work with emotion - at least I haven't found a working setup yet

**Reply by garronej** [reply]
That's a bummer,  

I'm not familiar with Astro but the main idea to make the SSR work is to collec the style from the emotion cache and insert them in the head.  

Here you have a low level example: https://emotion.sh/docs/ssr#advanced-approach   

With this we can theorycally make it work with any system... but it's not straight forward.  

Anyway, again, if you find a solution let me know so that I can make it available to everyone.  
Seem's like Astro is picking up steam.
